refactor(site): convert workspace batch delete dialog to Tailwind CSS

mafredri opened the pull request:
Converts from Emotion to Tailwind CSS, based on the tasks batch delete dialog implementation.

Also propagates simplifications back to the tasks dialog:
- Use `border-border` instead of hardcoded color variants
- Use `max-h-48` instead of specific `max-h-[184px]`
- Add cancel button to workspaces dialog

Refs #20905

---

Before:
<img width="1185" height="900" alt="capture-65fb658e" src="https://github.com/user-attachments/assets/8095f3af-40ec-4153-b643-ebcbf472cb0e" />

After:
<img width="1185" height="900" alt="capture-eea9631e" src="https://github.com/user-attachments/assets/4a94d1f6-3c9c-4f89-a8f5-0f5ea235449e" />

Before:
<img width="1185" height="900" alt="capture-cc670afa" src="https://github.com/user-attachments/assets/587e3ddf-0e90-46d3-9555-17540960d4af" />

After:
<img width="1185" height="900" alt="capture-a4453e13" src="https://github.com/user-attachments/assets/3f486ea2-cc1a-4b3a-b2e1-509316971fc4" />

mafredri committed:
refactor(site): swap emotion for tailwind in workspace batch delete dialogue

References #20905

mafredri committed:
fix button

jaaydenh reviewed (changes requested):
Avoid using `border-zinc-200 dark:border-zinc-700 `

mafredri committed:
fix(site): use theme border color and standard max-height in batch delete dialogs

- Replace border-zinc-200 dark:border-zinc-700 with border-border
- Replace max-h-[184px] with max-h-48 utility class
- Add cancel button to workspaces batch delete dialog

mafredri commented on site/src/pages/WorkspacesPage/BatchDeleteConfirmation.tsx:
```diff
@@ -150,69 +145,47 @@ const Workspaces: FC<StageProps> = ({ workspaces }) => {
 	return (
 		<>
-			<ul css={styles.workspacesList}>
+			<ul className="list-none p-0 border border-solid border-zinc-200 dark:border-zinc-700 rounded-lg overflow-x-hidden overflow-y-auto max-h-[184px]">
```
The reason I asked is that I saw the batch dialogue had a big revamp in look/flow. I can remove Stack if preferred.
